low-level/files/opt/seagate/sspl/bin/sspl_setup_init.py

```diff
@@ -0,0 +1,174 @@
+
+        # Check for dependency rpms and required processes active state based on role
+        if role == "ssu":
+            print(f"Checking for dependency rpms for role '{role}'")
+            self.check_for_dep_rpms(self.SSU_DEPENDENCY_RPMS)
+            print(f"Checking for required processes running state for role '{role}'")
+            self.check_for_active_processes(self.SSU_REQUIRED_PROCESSES)
+        elif role == "vm" or role == "gw" or role == "cmu":
+            print(f"Checking for dependency rpms for role '{role}'")
+            # No dependency currently. Keeping this section as it may be
+            # needed in future.
+            self.check_for_dep_rpms(self.VM_DEPENDENCY_RPMS)
+            # No processes to check in vm env
+        else:
+            print(f"No rpm or process dependencies set, to check for supplied role {role}, skipping checks.\n")
+
+    def get_uid(self, proc_name : str) -> int:
+        uid = -1
+        try :
+            uid = pwd.getpwnam(proc_name).pw_uid
+        except KeyError :
+            print(f"No such User: {proc_name}")
+        return uid
+
+    def process(self):
+        if not self.args:
+            self.usage()
+
+        i =0
+        while i < len(self.args):
+            if self.args[i] == '-dp':
+                # Extract the data path
+                sspldp = ''
+                with open(file_store_config_path, mode='rt') as confile:
+                    for line in confile:
+                        if line.find('data_path') != -1:
+                            sspldp = line.split('=')[1]
+                            break
+                # Crete the directory and assign permissions
+                try:
+                    # what should be the permissions for this directory ??
+                    os.makedirs(sspldp, mode=0o766, exist_ok=True)
+                    sspl_ll_uid = self.get_uid('sspl-ll')
+                    if sspl_ll_uid == -1:
+                        sys.exit(1)
+                    os.chown(sspldp, sspl_ll_uid, -1)
+                    for root, dirs, files in os.walk(sspldp):
+                        for item in dirs:
+                            os.chown(os.path.join(root, item), sspl_ll_uid, -1)
+                        for item in files:
+                            os.chown(os.path.join(root, item), sspl_ll_uid, -1)
+                except OSError as error:
+                    print(error) 
+
+            elif self.args[i] == '-r':
+                i+=1
+                if i>= len(self.args) or self.args[i] not in roles:
+                    self.usage()
+                else:
+                    self.role = self.args[i]
+            else:
+                self.usage()
+            i+=1
+        
+        # Create /tmp/dcs/hpi if required. Not needed for '<product>' role
+        if self.role != "cortx":
+            try:
+                os.makedirs('/tmp/dcs/hpi', mode=0o777, exist_ok=True)
```

Comment:
all these file/paths put in a var at the top & use it across, for code reusability

Reply:
done..